### PR TITLE
Ft tag token request test

### PR DESCRIPTION
**Pull request by andela-oojewale**

### CODE REVIEW

File: app/controllers/tokens_controller.rb [hunk 1]
```diff
@@ -1,10 +1,11 @@
-class TokensController < ApplicationController
+
+ class TokensController < ApplicationController
   before_action :set_token, only: [:validate]
   skip_before_action :authenticate_user, only: [:validate]
 
   def validate
     @user = @token.user
-    @user.update(active: true)
+    render json: {}, status: 200 if @user.update(active: true)
```
Comment (andela-oakinniranye) [inline app/controllers/tokens_controller.rb]: there's a jBuilder view that renders the json object for this, I guess. calling `render json: {}` just renders an empty object, whereas it ought to return the user's detail with token

File: db/schema.rb [hunk 2]
```diff
@@ -17,6 +17,7 @@
     t.integer  "user_id"
     t.integer  "question_id"
     t.string   "content"
+    t.integer  "votes"
```
Comment (andela-oakinniranye) [inline db/schema.rb]: There's a votes table that handles the votes so we can easily manage the votes, I doubt we'd be needing this. same for `comments and questions`

File: spec/models/tag_spec.rb [hunk 3]
```diff
@@ -4,9 +4,7 @@
   let(:arg) { :recent }
   let(:tag_name) { "Amity" }
   before :each do
-    create_list(:tag, 5, name: "Contract", created_at: 4.days.ago)
-    create_list(:tag, 3, name: "Amity", created_at: 2.days.ago)
-    create_list(:tag, 2, name: "Kaizen")
+    create_tags
```
Comment (andela-oakinniranye) [inline spec/models/tag_spec.rb]: where was this method defined? 💭

File: spec/requests/tags_request_spec.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
+require 'rails_helper'
+
+RSpec.describe TagsController, type: :request do
+
+  let(:user) { valid_user }
+  let(:contact_tag) { Tag.find_by(name: "contract") }
```
Comment (andela-oakinniranye) [inline spec/requests/tags_request_spec.rb]: i think this ought to be `contract_tag`

File: spec/requests/tags_request_spec.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  before :each do
+    create_tags
+  end
+
+  describe "GET /tags" do
+    it "returns 200 status and tag" do
+      allow(Tag).to receive(:search).and_return([contact_tag])
+      get tags_path("contract"), {format: :json}, authorization_header
```
Comment (andela-oakinniranye) [inline spec/requests/tags_request_spec.rb]: I think you mean: 
```ruby
get tags_path(q: 'contract'), {format: :json}, authorization_header
```

File: spec/requests/tags_request_spec.rb [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+      get popular_tags_path, {format: :json}, authorization_header
+
+      expect(parsed_json).to have_key("contract")
+      expect(status).to eq 200
+    end
+  end
+
+  describe "GET /tags/recent" do
+    it "returns 200 status and recent tags" do
```
Comment (andela-oakinniranye) [inline spec/requests/tags_request_spec.rb]: I think the block names should be more explicit. something like `returns status 200 with all tags ordered by the recently created ones`

File: spec/support/json_parser_helper.rb [hunk 7]
```diff
@@ -6,6 +6,10 @@ def parsed_json
   def body
     response.body
   end
+
+  def status
+    response.status
+  end
```
Comment (andela-oakinniranye) [inline spec/support/json_parser_helper.rb]: nice 👍

File: spec/support/create_resource.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+  end
+
+  def create_user
+    create(:user)
+  end
+end
+
+RSpec.configure do |config|
+  config.include CreateResource
+end
```
Comment (andela-oakinniranye) [inline spec/support/create_resource.rb]: nice work here 👍

File: spec/requests/tags_request_spec.rb (hunk 5 above)
Comment (andela-oojewale) [inline spec/requests/tags_request_spec.rb]: That defaults to the `q` though. I correct if for clarity.

File: spec/requests/tags_request_spec.rb (hunk 4 above)
Comment (andela-oojewale) [inline spec/requests/tags_request_spec.rb]: True. Thanks

File: spec/requests/tags_request_spec.rb (hunk 6 above)
Comment (andela-oojewale) [inline spec/requests/tags_request_spec.rb]: Cool

File: spec/support/json_parser_helper.rb (hunk 7 above)
Comment (andela-oojewale) [inline spec/support/json_parser_helper.rb]: Thanks.

File: spec/support/create_resource.rb (hunk 8 above)
Comment (andela-oojewale) [inline spec/support/create_resource.rb]: Thanks

File: spec/requests/tags_request_spec.rb (hunk 5 above)
Comment (andela-oojewale) [inline spec/requests/tags_request_spec.rb]: I just found out that was wrong. Pardon me.

File: spec/rails_helper.rb [hunk 9]
```diff
@@ -12,8 +12,8 @@
 require "shoulda/matchers"
 require "spec_helper"
 require "rspec/rails"
-require 'coveralls'
-Coveralls.wear!
+# require 'coveralls'
+# Coveralls.wear!
```
Comment (andela-echigbo) [inline spec/rails_helper.rb]: I think you should uncomment this, to give report to `coveralls`

File: app/controllers/tags_controller.rb [hunk 10]
```diff
@@ -18,5 +18,6 @@ def trending
 
   def update_subscription
     current_user.tags = Tag.process_tags(params[:tags])
+    render json: {}, status: 200
```
Comment (andela-oakinniranye) [inline app/controllers/tags_controller.rb]: I know it's not meant to be your task, but could you help write a jbuilder to return all the tags the user is now subscribed to?

**Pull request merged**